bundles/org.openhab.binding.tsmart/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,102 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="tsmart"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="https://openhab.org/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
+
+	<thing-type id="tsmart">
+
+		<label>Tesla T-Smart</label>
+		<description>Tesla T-Smart Immersion Heater Thermostat</description>
+
+		<channels>
+			<channel id="power" typeId="power"/>
+			<channel id="temperature" typeId="temperature">
+				<label>Temperature</label>
+				<description>Measured temperature across the sensors</description>
+			</channel>
+			<channel id="temperatureHigh" typeId="temperature">
```

Comment:
Please adapt all channel ids and channel type ids according to [naming convention]:(https://www.openhab.org/docs/developer/guidelines.html#naming-convention)
```suggestion
			<channel id="temperature-high" typeId="temperature">
```